### ISSUE TITLE
Related items widget searches only on depth 1, not -1

### ISSUE THREAD

**Issue by malthe** [plone/Products.CMFPlone#1986]
### What I did:

The `plone.app.z3cform.widget.RelatedItemsWidget` has a pattern option `pathDepth` that affects both browsing and searching.

If you set it to -1 then tree browsing makes no sense. If you set it to 1 (the default) then searching makes no sense.

### What I expect to happen:

I would expect that I can browser for a folder and then search below that folder (any depth).

### What actually happened:

You always search only in the selected folder. For example, if you click on the house, typically nothing is shown.

### What version of Plone/ Addons I am using:

Plone 5.

**Comment by thet** [comment]
there is a PR which might solve your problem:
https://github.com/plone/mockup/pull/752

this PR implements a new mode "auto". it switches to search mode automatically, when a search term is present, otherwise it's in browse mode. in search mode, the search is done on the current path with any depth.

I suggest to remove the ``pathDepth`` option...

**Comment by thet** [comment]
... i can't find the ``pathDepth`` option, so there might be no need to remove it. where did you find it, @malthe ?

**Comment by malthe** [comment]
It's buried deep in the "thememapper-compiled.js" file. I haven't been able to trace another source for it.

Excerpt:
```javascript
    var defaults = {
      pattern: null, // must be passed in
      vocabularyUrl: null,
      searchParam: 'SearchableText', // query string param to pass to search url
      attributes: ['UID', 'Title', 'Description', 'getURL', 'portal_type'],
      batchSize: 10, // number of results to retrive
      baseCriteria: [],
      sort_on: 'is_folderish',
      sort_order: 'reverse',
      pathDepth: 1
    };
```

**Comment by malthe** [comment]
On another note, it seems like someone had a field day coming up with a spider web of interdependent packages, `plone.app.z3cform`, `plone.app.theming`, `plone.app.vocabularies`, ...

It was remarkably difficult to get to the root of this issue. I had not expected to find the problem in `plone.app.theming`.

My solution ended up being:
```python
class AnyDepthCatalogSource(CatalogSource):
    def search_catalog(self, user_query):
        query = user_query.get('path')
        if query is not None:
            query = query.copy()
            query['depth'] = -1
            user_query = user_query.copy()
            user_query['path'] = query
        return user_query
```
That is hardly ideal.

The problem with this sort of API is that `search_catalog` basically gets no chance to impart its own policy because the argument is just a query dictionary with more or less arbitrary content.

**Comment by thet** [comment]
It's an issue with mockup, which is already fixed in Plone 5.1 and improved in my PR.
I suggest to use mockup master and wait for my PR to be merged (tests have to be fixed, shouldn't be a big deal).

So, to reconstruct your problem.... In Plone 5.0, we use the mockup 2.1.x branch.

``pathDepth`` is used here: https://github.com/plone/mockup/blob/2.1.x/mockup/js/utils.js#L10 - see Line 27 and 109.
If you set is as an option on the related items widget, it is passed to mockup.js.utils.QueryHelper.
But ``pathDepth`` is only set when in browsing mode or when ``searchPath`` is present. But ``searchPath`` isn't used in the related items widget, so actually in search mode the whole site should be searched.

The mode wasn't set automatically in Plone 5.0. It was set via pattern options. The default was ``search``.

In 5.1 (which uses mockup master) users could choose via search and browse mode by pressing a button.

In my PR, the ``auto`` mode is introduced, which automatically sets the mode based on the presence of a search term.

**Comment by thet** [comment]
Fixed in Plone 5.1 via the merge of https://github.com/plone/mockup/pull/752
